core/src/main/java/io/questdb/griffin/engine/functions/window/DenseRankFunctionFactory.java

```diff
@@ -0,0 +1,359 @@
+        }
+
+        @Override
+        public void initRecordComparator(RecordComparatorCompiler recordComparatorCompiler, ArrayColumnTypes chainTypes, IntList order) {
+            this.recordComparator = recordComparatorCompiler.compile(chainTypes, order);
+        }
+
+        @Override
+        public void pass1(Record record, long recordOffset, WindowSPI spi) {
+            partitionByRecord.of(record);
+
+            MapKey mapKey = map.withKey();
+            mapKey.put(partitionByRecord, partitionBySink);
+            MapValue mapValue = mapKey.createValue();
+            long maxIndex = 0;
+            if (mapValue.isNew()) {
+                mapValue.putLong(VAL_MAX_INDEX, 0);
+                mapValue.putLong(VAL_CURRENT_INDEX, 0);
+                mapValue.putLong(VAL_OFFSET, 0);
+            } else {
+                maxIndex = mapValue.getLong(VAL_MAX_INDEX);
+            }
+
+            if (recordComparator == null) {
+                // no order or order dismiss
+                Unsafe.getUnsafe().putLong(spi.getAddress(recordOffset, columnIndex), maxIndex + 1);
+            } else {
+                long currentIndex = mapValue.getLong(VAL_CURRENT_INDEX);
+                long offset = mapValue.getLong(VAL_OFFSET);
+                if (currentIndex == 0) {
+                    mapValue.putLong(VAL_CURRENT_INDEX, 1);
+                    mapValue.putLong(VAL_OFFSET, recordOffset);
+                } else {
+                    // compare with prev record
+                    recordComparator.setLeft(record);
+                    if (recordComparator.compare(spi.getRecordAt(offset)) != 0) {
+                        mapValue.putLong(VAL_CURRENT_INDEX, maxIndex + 1);
+                        mapValue.putLong(VAL_OFFSET, recordOffset);
+                    }
+                }
+                Unsafe.getUnsafe().putLong(spi.getAddress(recordOffset, columnIndex), mapValue.getLong(VAL_CURRENT_INDEX));
+            }
+            mapValue.putLong(VAL_MAX_INDEX, maxIndex + 1);
+        }
+
+        @Override
+        public void reopen() {
+            map.reopen();
+        }
+
+        @Override
+        public void reset() {
+            map.close();
+        }
+
+        @Override
+        public void setColumnIndex(int columnIndex) {
+            this.columnIndex = columnIndex;
+        }
+
+        @Override
+        public void toPlan(PlanSink sink) {
+            sink.val(SIGNATURE);
+            sink.val(" over (");
+            sink.val("partition by ");
+            sink.val(partitionByRecord.getFunctions());
+            sink.val(')');
+        }
+
+        @Override
+        public void toTop() {
+            map.clear();
+        }
+    }
+
+    private static class SequenceDenseRankFunction extends LongFunction implements ScalarFunction, WindowFunction, Reopenable {
+
+        private int columnIndex;
+
+        private long rank;
+
```

Comment:
Constructor should match class name:  SequenceRankFunction -> SequenceDenseRankFunction .